src/yoroi-wallets/walletManager.ts

```diff
@@ -417,12 +428,12 @@ export class WalletManager {
   // returns the corresponding implementation of WalletInterface. Normally we
   // should expect that each blockchain network has 1 wallet implementation.
   // In the case of Cardano, there are two: Byron-era and Shelley-era.
-  _getWalletImplementation(walletImplementationId: WalletImplementationId): WalletInterface {
+  _getWalletImplementation(walletImplementationId: WalletImplementationId): typeof ShelleyWallet {
```

Comment:
this doesnt return `typeof ShelleyWallet` 
it returns `ShelleyWallet`
but the return type should be inferred, anyway